fitbenchmarking/fitbenchmarking
# Options printing

#### Description of Work

Fixes #663 
Fixes #658 

Adds options printing to `DEBUG` log level and prints warning if `options.result_dir` is set to be the default.

#### Testing Instructions

1. Check options printing works for `DEBUG` log level
2. Check the wording of the warning for the results directory

Function: Does the change do what it's supposed to?

Tests: Does it pass? Is there adequate coverage for new code?

Style: Is the coding style consistent? Is anything overly confusing?

Documentation: Is there a suitable change to documentation for this change?
